### PR TITLE
commit api

### PR DESCRIPTION
**Pull request by mikiquantum**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: httpapi/router.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 // Router returns the http mux for the server.
 // @title Centrifuge OS Node API
 // @description Centrifuge OS Node API
-// @version 0.0.5
+// @version 0.0.6
```
Comment (vedhavyas) [inline httpapi/router.go]: why the change in version? Did we break anything from the old APIs

File: httpapi/v2/documents_api.go [hunk 2]
```diff
@@ -72,3 +74,49 @@ func (h handler) CreateDocument(w http.ResponseWriter, r *http.Request) {
...
+// @id commit_document_v2
+// @tags Documents
+// @accept json
+// @param authorization header string true "Hex encoded centrifuge ID of the account for the intended API action"
+// @param document_id path string true "Document Identifier"
+// @produce json
+// @Failure 400 {object} httputils.HTTPError
+// @Failure 500 {object} httputils.HTTPError
+// @Failure 403 {object} httputils.HTTPError
+// @success 201 {object} coreapi.DocumentResponse
```
Comment (vedhavyas) [inline httpapi/v2/documents_api.go]: the status should be accepted 202. WDYT ?

File: pending/test_pending.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ package pending
 import (
 	"context"
 
+	"github.com/centrifuge/go-centrifuge/jobs"
+
```
Comment (vedhavyas) [inline pending/test_pending.go]: import space

File: testworld/v2_documents_test.go [hunk 4]
```diff
@@ -51,3 +51,28 @@ func TestV2DocumentCreate_next_version(t *testing.T) {
 	// alice should not have this version
 	nonExistingDocumentVersionCheck(alice.httpExpect, alice.id.String(), "documents", params)
```
Comment (vedhavyas) [inline testworld/v2_documents_test.go]: can you adapt commit to the functions NewDocument and Exisitng document?
We already have created a pending document at this point instead of re-creating a new one ?

Comment (mikiquantum) [reply testworld/v2_documents_test.go]: I rather keep it separate since it is a separate API call independent of the other tests. Easier to track.

File: httpapi/v2/documents_api.go (hunk 2 above)
Comment (mikiquantum) [reply httpapi/v2/documents_api.go]: yes!

File: httpapi/router.go (hunk 1 above)
Comment (mikiquantum) [reply httpapi/router.go]: 0.0.5 is what we released as part of Eunomia, and we have been overwritting all over, since the release (pushing to swaggerhub). We need to be more careful when cutting the release to change the version on the first commit after.

File: testworld/v2_documents_test.go (hunk 4 above)
Comment (vedhavyas) [reply testworld/v2_documents_test.go]: the new test is making the same API calls as the new one. Adapting to old tests ensures we do not have a lot of tests doing the same. WDYT ?
